### ISSUE TITLE
Flow errors about State type in combination with Redux and flow-typed

### ISSUE THREAD

**Issue by Strayer** [apollographql/react-apollo#898]
**Intended outcome:**
react-apollo should work with react-redux, redux and flow-typed out of the box.

**Actual outcome:**
Running the flow type validation shows type errors in react-apollo related to redux Store type.

```
node_modules/react-apollo/index.js.flow:30
 30:     store: Store,
                ^^^^^ Store. Application of polymorphic type needs <list of 2 arguments>. (Can use `*` for inferrable ones)
```

**How to reproduce the issue:**
Reproduced in this repo: https://github.com/Strayer/ApolloFlowIssues
Install dependencies, then run `yarn run flow` to see the errors.

**Version**
- apollo-client@1.9.1
- react-apollo@1.4.12
- react-redux@5.0.6
- redux@3.7.2

Edit 2017-08-11: Removed now invalid part about import errors, updated versions.

**Comment by stevemolitor** [comment]
I am seeing the same errors. If I change for example `store: Store` to `store: Story<any>` in the react-apollo flow declarations, the errors go away.

**Comment by barbu110** [comment]
Anything on this yet?

**Comment by Strayer** [comment]
The import errors seem to be resolved in react-apollo@1.4.12, the errors about store are still occurring though. Updated the example project to the newest versions.
